tests/func/test_api.py

```diff
@@ -114,6 +114,10 @@ def test_missing(remote_url, tmp_dir, dvc):
     # Remove cache to make foo missing
     remove(dvc.cache.local.cache_dir)
 
+    api.read("foo")
```

Comment:
There is no reason for this to not work. E.g. in a described --no-exec/--no-commit scenario.